### ISSUE TITLE
IsCellActive check doesn't work properly

### ISSUE THREAD

**Issue by einar90** [PetroleumCyberneticsGroup/FieldOpt#78]
The check to see whether a cell is active or not does not work properly. It returns `true` for a lot of inactive cells.

**Comment by agchitu** [comment]
Hi Einar,

I am currently looking at this myself these days.... I do not know whether you did something about it or not. I'll create a WIP PR to let you see my progress. My current interest is to add functionality that allows for dual grids. The way we check for active cells will be changed since it depends on which grid the cell is active in. They can be active in the matrix grid but not in the fracture grid.
Alin

**Comment by einar90** [comment]
Hi Alin, 

Sorry for the late reply, Mathias and I are working from California nowadays. 

We've been looking into it, but we haven't managed to solve it. I believe we have to take multiple grids (local grids) into consideration for all non-trivial reservoirs, at least that's what it looks like (the current check doesn't even work in the Brugge model, which shouldn't be that complicated), so you're probably on the right track for fixing the problem in general.

ResInsight (an open source visualization program that we use) also uses ERT to read the grid, and it seems they use local grid indices when checking if a cell is active. [This](https://github.com/OPM/ResInsight/blob/d3f512f783298c513d871411ccb8d8b598ec3eef/ApplicationCode/FileInterface/RifReaderEclipseOutput.cpp#L98) may be a good place to start.

We're also looking into using ACTNUM from the INIT file directly to check if a cell is active. This may or may not be a simpler solution, we're not sure yet.

Einar

**Comment by agchitu** [comment]
Hi Einar, 
As I wrote before and you will see in the WIP PR I am working on this and I have made good progress already. I'll share more code soon (I am only working on Fridays on this project). One thing I found already is that there is a bug in the current version of our code in the way we were using the IsCellActive. This method takes the global index as argument and we were using it with the active index. This might explain why some cells are found as inactive when we expect them to be active. I have worked to improve other places as well but that will come with the PR. 

Do you guys use the new ert project or you still use the old one? I am still using the old one but I was advised to move to the new one soon. At this moment there should not be any functionality change in libecl but there might be at some point. 

Cheers, Alin

**Comment by einar90** [comment]
Hi Alin,
Sorry for the late reply, I just got back from the states. I'll take a look at your PR today and merge it if the tests pass; this fix is very needed so I want to get it in as quickly as possible.

I've switched to the new ERT project on my laptop, doesn't look like there are any significant changes, but I'm thinking it might be a good idea to fork the the two libraries we use and include them as subrepos and subdirectories (in CMake); this should make installation easer (seeing that one wouldn't have to install ERT, it would be compiled along with FieldOpt). 

BTW, I'm going on vacation this friday and I won't be back until the second week of august. And my laptop is broken. So if I'm not answering your comments the next month, that's why (send me an email if there's something important).

Einar
